contrib/src/main/java/com/datatorrent/contrib/parser/CsvParser.java

```
@@ -62,247 +71,193 @@
+      }
+      errorTupleCount++;
+      return;
     }
+    try {
+      if (parsedOutput.isConnected()) {
+        csvStringReader.open(incomingString);
+        Map<String, Object> map = csvMapReader.read(nameMapping, processors);
+        parsedOutput.emit(map);
+        parsedOutputCount++;
+      }
 
-    public String getFormat()
-    {
-      return format;
-    }
+      if (out.isConnected() && clazz != null) {
+        csvStringReader.open(incomingString);
+        Object obj = csvBeanReader.read(clazz, nameMapping, processors);
+        out.emit(obj);
+        emittedObjectCount++;
+      }
 
-    public void setFormat(String format)
-    {
-      this.format = format;
+    } catch (SuperCsvException e) {
+      if (err.isConnected()) {
+        err.emit(new KeyValPair<String, String>(incomingString, e.getMessage()));
+      }
+      errorTupleCount++;
+      logger.error("Tuple could not be parsed. Reason {}", e.getMessage());
+    } catch (IOException e) {
+      logger.error("Exception in process method {}", e.getMessage());
+      DTThrowable.rethrow(e);
     }
-
-  }
-
-  /**
-   * Gets the array list of the fields, a field being a POJO containing the name
-   * of the field and type of field.
-   * 
-   * @return An array list of Fields.
-   */
-  public ArrayList<Field> getFields()
-  {
-    return fields;
-  }
-
-  /**
-   * Sets the array list of the fields, a field being a POJO containing the name
-   * of the field and type of field.
-   * 
-   * @param fields
-   *          An array list of Fields.
-   */
-  public void setFields(ArrayList<Field> fields)
-  {
-    this.fields = fields;
   }
 
-  /**
-   * Gets the delimiter which separates fields in incoming data.
-   * 
-   * @return fieldDelimiter
-   */
-  public int getFieldDelimiter()
+  @Override
+  public KeyValPair<String, String> processErorrTuple(byte[] input)
   {
-    return fieldDelimiter;
+    //This method is not invoked for CSV parser
+    return null;
```

Please throw UnsupportedOperationException if this is not meant to be called.